dev/com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.impl/src/com/ibm/ws/microprofile/reactive/messaging/kafka/adapter/impl/ProducerRecordFactory.java

```diff
@@ -61,4 +61,39 @@
 
         return delegateRecord;
     }
+
+    /**
+     * Extract a suitable delegate ProducerRecord from a user provided one.
+     * It will be a brand new ProducerRecord if the configuredTopic does not match the one in the user ProducerRecord.
+     * If the topic is the same then this method will just return the user's ProducerRecord as-is
+     *
+     * @param <K> key type
+     * @param <V> value type
+     * @param configuredTopic The configured topic
+     * @param userProducerRecord The user's producer record
+     * @return
+     */
+    public static <K, V> org.apache.kafka.clients.producer.ProducerRecord<K, V> extractUserProducerRecord(String configuredTopic,
+                                                                                                          org.apache.kafka.clients.producer.ProducerRecord<K, V> userProducerRecord) {
+
+        org.apache.kafka.clients.producer.ProducerRecord<K, V> delegateRecord;
+
+        String userTopic = userProducerRecord.topic();
+
+        //if a topic was not configured, use the one in the user provided ProducerRecord
+        String topic = configuredTopic == null ? userTopic : configuredTopic;
+        if (topic.contentEquals(userTopic)) {
```

Comment:
`if (configuredTopic == null || configuredTopic.equals(userProducerRecord.topic())) {` might be clearer here